### ISSUE TITLE
activate script is broken

### ISSUE THREAD

**Issue by asmeurer** [conda/conda#2691]
```
$act lifelines
prepending /Users/aaronmeurer/anaconda/envs/lifelines/bin to PATH
bash: /usr/bin/sed: Argument list too long
bash: /Users/aaronmeurer/anaconda3/bin/conda: Argument list too long
bash: /bin/sh: Argument list too long
bash: /bin/sh: Argument list too long
```

After this, my `$PS1` is completely wiped, and all further bash commands give the same "argument list is too long" error. 

Downgrading to conda 4.0.8 fixes it. 

Before it did this and was completely broken, I also noticed that `CONDA_DEFAULT_ENV` was giving the full path to the env, instead of the short version, which broke my custom `PS1`. Should I open a separate issue for that?

**Comment by kalefranz** [comment]
CC @msarahan 

Aaron you might have some old files lingering in your environment.  Like if you installed conda from source with setup.py before.  That was the solution for me yesterday when I ran into this and @msarahan helped me clear it up.

**Comment by msarahan** [comment]
CONDA_DEFAULT_ENV is always going to show the full path now (unless Kale says otherwise).  It was a mess to have it be either the full path or just the name.  I'd be OK with some other way for you to get just the name, though.  @kalefranz, we could have the output of ..setps1 be just the short name, and add whatever parentheses or brackets we wanted in the shell scripts, as well as add on the original prompt.

**Comment by asmeurer** [comment]
The `CONDA_DEFAULT_ENV` thing is a pretty serious regression. A lot of people (including myself) have put it in PS1 as a replacement for the default PS1 handling. Why can't you have a new separate environment variable for the full path?

**Comment by asmeurer** [comment]
Any idea what sort of files I should be looking for? I don't think I've ever installed conda from source on this computer, but I could be misremembering.

**Comment by kalefranz** [comment]
So @asmeurer previous to 4.1, `CONDA_DEFAULT_ENV` has just been `basename $PREFIX`?

**Comment by msarahan** [comment]
No, it was inconsistent.

**Comment by asmeurer** [comment]
It was the shortest name such that `activate $CONDA_DEFAULT_ENV` activated the environment. So if the env was in one of the default envs directories, it was the env directory name. Otherwise, it was the full path.

**Comment by megies** [comment]
I'm experiencing a problem with `4.1.x` that seems related to me.

Since updating on one computer, activation of root environment takes forever (>1min) and then I get some funky errors like..
```
/home/obspy/anaconda3/bin/activate: line 66: sed: command not found
git: command not found
rm: command not found
```

I've looked at `PATH` and it looks thoroughly mangled:
```
((root)) obspy@h2114090:~$ conda install -n root --force conda
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /home/obspy/anaconda3:

The following NEW packages will be INSTALLED:

    conda: 4.1.2-py35_0

Proceed ([y]/n)? 

Pruning extracted packages from the cache ...
[      COMPLETE      ]|#####################################################################################################| 100%
Extracting packages ...
[      COMPLETE      ]|#####################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|#####################################################################################################| 100%
((root)) obspy@h2114090:~$ conda update conda
Error: Package 'conda' is not installed in /home/obspy/anaconda3/envs/
((root)) obspy@h2114090:~$ source deactivate
-bash: dirname: command not found
-bash: uname: command not found
obspy@h2114090:~$ echo $PATH
/home/obspy/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
(reverse-i-search)`source': conda clean --packages --^Curce-cache --index-cache --tarballs
obspy@h2114090:~$ . ~/anaconda3/bin/activate root
prepending /home/obspy/anaconda3/bin to PATH
((root)) obspy@h2114090:~$ echo $PATH
/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/anaconda3/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/home/obspy/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
```
(note also the confusing messages about `conda` not being installed.. O_o)

For me too, downgrading to conda 4.0.8 fixes it.

**Comment by msarahan** [comment]
Yeah, sorry, you are seeing #2699.  I think this is fixed by #2831 - but haven't seen the CI results yet. Hopefully we can get a fixed release out very soon.

**Comment by megies** [comment]
Ah.. did a quick search through issues, but didn't stumble over that one.. thanks for the info.
